service/src/main/kotlin/no/nav/su/se/bakover/service/AccessCheckProxy.kt

```diff
@@ -0,0 +1,314 @@
+                    assertHarTilgangTilSak(sakId)
+
+                    return services.sak.hentSak(sakId)
+                }
+
+                override fun hentSak(fnr: Fnr): Either<FantIkkeSak, Sak> {
+                    assertHarTilgangTilPerson(fnr)
+
+                    return services.sak.hentSak(fnr)
+                }
+
+                override fun opprettSak(sak: NySak) {
+                    assertHarTilgangTilPerson(sak.fnr)
+
+                    return services.sak.opprettSak(sak)
+                }
+            },
+            søknad = object : SøknadService {
+                override fun nySøknad(søknadInnhold: SøknadInnhold): Either<KunneIkkeOppretteSøknad, Sak> {
+                    assertHarTilgangTilPerson(søknadInnhold.personopplysninger.fnr)
+
+                    return services.søknad.nySøknad(søknadInnhold)
+                }
+
+                override fun hentSøknad(søknadId: UUID): Either<KunneIkkeLukkeSøknad.FantIkkeSøknad, Søknad> {
+                    assertHarTilgangTilSøknad(søknadId)
+
+                    return services.søknad.hentSøknad(søknadId)
+                }
+
+                override fun trekkSøknad(
+                    søknadId: UUID,
+                    saksbehandler: NavIdentBruker.Saksbehandler,
+                    begrunnelse: String
+                ): Either<KunneIkkeLukkeSøknad, Sak> {
+                    assertHarTilgangTilSøknad(søknadId)
+
+                    return services.søknad.trekkSøknad(søknadId, saksbehandler, begrunnelse)
+                }
+            },
+            brev = object : BrevService {
+                override fun journalførVedtakOgSendBrev(
+                    sak: Sak,
+                    behandling: Behandling
+                ): Either<KunneIkkeOppretteJournalpostOgSendeBrev, String> {
+                    assertHarTilgangTilPerson(sak.fnr)
+
+                    return services.brev.journalførVedtakOgSendBrev(sak, behandling)
+                }
+
+                override fun lagBrev(request: LagBrevRequest): Either<KunneIkkeLageBrev, ByteArray> {
+                    assertHarTilgangTilPerson(request.getFnr())
+
+                    return services.brev.lagBrev(request)
+                }
+
+                override fun journalførBrev(
+                    request: LagBrevRequest,
+                    sakId: UUID
+                ): Either<KunneIkkeJournalføreBrev, JournalpostId> {
+                    assertHarTilgangTilSak(sakId)
+
+                    return services.brev.journalførBrev(request, sakId)
+                }
+
+                override fun distribuerBrev(journalpostId: JournalpostId): Either<KunneIkkeSendeBrev, String> {
+                    return services.brev.distribuerBrev(journalpostId)
+                }
+            }
+        )
+    }
+
+    private fun assertHarTilgangTilPerson(fnr: Fnr) {
+        clients.personOppslag.person(fnr)
+            .getOrHandle { throw Tilgangssjekkfeil(it, fnr) }
```

Comment:
Kan ikke PersonOppslag returnere feiltyper vi ikke vil kaste Tilgangssjekkfeil på?

Reply:
Tja, den kan returnere andre ting enn `IkkeTilgangTilPerson`, men jeg tenker at siden vi ønsker å legge oss på linja med "deny by default" så er det bedre at vi bare kaster her dersom vi ikke kan fastslå at bruker har tilgangen.

Reply:
Jeg synes i utgangspunktet ikke det er en Tilgangssjekkfeil's ansvar og formidle at et fnr ikke finnes. Nå har vi bare 1 inngang som tar imot fnr og ikke sak tror jeg. Men der burde vi kunne skille på at brukeren ikke har tilgang vs. fødselsnummeret er ikke i bruk? Det kan jo løses spesifikt for det tilfellet.

Reply:
Kan være enig i det, men per nå har vi vel ikke disse potensielle mekanismene inne. Hva tenker du om at vi lar dette være som det er for nå og så spisser vi det heller senere når vi lager bedre håndtering av det?